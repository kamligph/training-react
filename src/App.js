import React, { Component } from 'react';
import Projects from './Component/Projects';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        Training React App
        <Projects />
      </div>
    );
  }
}

export default App;
