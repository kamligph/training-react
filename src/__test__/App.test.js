import React from 'react'; 
import axios from 'axios';
import sinon from 'sinon';
import renderer from 'react-test-renderer';
import { mount, shallow, render } from 'enzyme';
import App from './../App'; 
import ReactTestUtils from 'react-dom/test-utils';

describe('[App.js]', () => {
  it('1. Renders without crashing', () => {
    const wrapper = renderer.create(
      <App />
    );

    const tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('2. Calls componentDidMount', () => {
    sinon.spy(App.prototype, 'componentDidMount');
    const wrapper = mount(<App />);
    expect(App.prototype.componentDidMount.calledOnce).toBe(true);
  });

  it('3. Show form elements', () => {
    const wrapper = mount(
      <App />
    );

    const form = wrapper.find('form');
    expect(form.length).toBe(1);
    expect(form.find('input[type|="text"]').length).toBe(1);
    expect(form.find('select').length).toBe(1);
    expect(form.find('input[type|="submit"]').length).toBe(1);
  });

  it('4. Shows 3 projects', async() => {
    const wrapper = shallow(
      <App />
    );

    expect.assertions(1);
    await wrapper.instance().componentDidMount();
    expect(wrapper.state('projects').length).toBe(3);
  });

  it('5. Get promised todos v1', async() => {    
    expect.assertions(1);
    await axios.get('https://jsonplaceholder.typicode.com/todos')
      .then((response) => {
        expect(response.data.length).toBe(200);
      })
  });

  it('6. Get promised todos v2', () => {

    const onButtonClick = sinon.spy();
    const wrapper = render(<App onButtonClick={onButtonClick} />);
    // wrapper.find('input[type|="submit"]').simulate('click');
    expect(wrapper[0].children).toBe(1);
    // expect(onButtonClick).toHaveProperty('callCount', 1);

    // # working
    // const wrapper = mount(<App bar="baz" />);
    // expect(wrapper.props().bar).toEqual('baz');
    // wrapper.setProps({ bar: 'foo' });
    // expect(wrapper.props().bar).toEqual('foo');

    // var f = sinon.fake();
    // var date1 = new Date();
    // var date2 = 'fsfad';

    // f('sfsfs', date1);
    // f('a', 2, date2);

    // f.lastArg === date2;
    // expect(f.lastArg).toBe('fsfad');
    
    
    
    // expect.assertions(1);
    // await axios.get('https://jsonplaceholder.typicode.com/todos')
    //   .then((response) => {
    //     expect(response.data.length).toBe(200);
    //   })
    // const component = renderer.create(
    //   <App />
    // );

    // return ... then(data => {
    //   expect(data.name).toEqual('');
    // })
    
    // return App.getTodos().then(data => {
    //   expect(data.name).toEqual('shit shit');
    // })
  });

  // it('simulates click events', () => {
  //   const onButtonClick = sinon.spy();
  //   const wrapper = mount((
  //     <Foo onButtonClick={onButtonClick} />
  //   ));
  //   wrapper.find('button').simulate('click');
  //   expect(onButtonClick.calledOnce).to.equal(true);
  // });
});
