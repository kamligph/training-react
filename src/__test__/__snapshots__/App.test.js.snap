// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Web Application Renders without crashing 1`] = `
<div
  className="App"
>
  <div>
    <h3>
      Add Project
    </h3>
    <form
      onSubmit={[Function]}
    >
      <div>
        <label>
          Title
        </label>
        <br />
        <input
          type="text"
        />
      </div>
      <div>
        <label>
          Category
        </label>
        <br />
        <select>
          <option
            value="Web Design"
          >
            Web Design
          </option>
          <option
            value="Web Development"
          >
            Web Development
          </option>
          <option
            value="Mobile Development"
          >
            Mobile Development
          </option>
        </select>
      </div>
      <br />
      <input
        type="submit"
        value="Submit"
      />
    </form>
  </div>
  <div
    className="Projects"
  >
    <h3>
      Latest Projects
    </h3>
    <li
      className="Project"
    >
      <strong>
        Business Website
      </strong>
      : 
      Web Design
       
      <button
        onClick={[Function]}
      >
        x
      </button>
    </li>
    <li
      className="Project"
    >
      <strong>
        Social App
      </strong>
      : 
      Mobile Development
       
      <button
        onClick={[Function]}
      >
        x
      </button>
    </li>
    <li
      className="Project"
    >
      <strong>
        Ecommerce Cart
      </strong>
      : 
      Web Development
       
      <button
        onClick={[Function]}
      >
        x
      </button>
    </li>
  </div>
  <hr />
  <div
    className="Todos"
  >
    <h3>
      To Dos
    </h3>
  </div>
</div>
`;
